pkg/globalnet/controllers/cluster_egressip_controller_test.go

```diff
@@ -35,7 +37,15 @@ var _ = Describe("ClusterGlobalEgressIP controller", func() {
 	When("the well-known ClusterGlobalEgressIP does not exist on startup", func() {
 		It("should create it and allocate the global IP", func() {
 			t.awaitClusterGlobalEgressIPStatusAllocated(controllers.ClusterGlobalEgressIPName, 1)
-			// TODO - verify IP tables
+			// Verify that IPtable rule is programmed in SM-GN-EGRESS-CLUSTER chain with allocatedIPs
+			t.ipt.AwaitRule("nat", constants.SmGlobalnetEgressChainForCluster,
```

Comment:
This is covering functional testing which we should probably keep on a different category, and even different execution to unit tests. For example, I'm not sure if we could execute those tests properly on an OSX environment.

Can we separate this type of system-interacting test to a separate file at least

cluster_egressip_controller_functional_test.go

And may be prepend this type of test with:

Describe("[functional] ClusterGlobalEgressIP controller", func()

So later we could... either move them to a separate $project/test/functional or filter them out in ginkgo?

Reply:
As you probably know, we use fake iptables (pkg/iptables/fake) during the unit testing. We were doing similar validation even with the existing globalnet. I hope we did not find any issue running on OSX (please correct me if this is not the case). Basically, if we did not have any issue so far on OSX, we should not have any issue even with this PR as well.